### PR TITLE
Generate code for complianceoperator

### PR DESCRIPTION
**Pull request by janisz**
## Description

Although `zz_generated.deepcopy.go` claims to be autogenerated it's not generated by `make generate`.
This PR fixes that and also switch from `+k8s:deepcopy-gen:interfaces=...` to `+kubebuilder:object:root=true` as mentioned in [Operator-SDK Migration guide](https://github.com/operator-framework/operator-sdk/blob/62cbecc5744ca290046afff16333aec8043638f7/website/content/en/docs/building-operators/golang/migration.md#L125)   

## Testing Performed

CI

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2992** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2992" target="_blank">Graphite</a>  👈
    * **PR #2969** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/2969" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/2992?utm_source=stack-comment).

**Comment by janisz** [comment]
@porridge @connorgorman PTAL

File: operator/Makefile
Comment (porridge) [inline operator/Makefile:244]: I don't think it's a good idea to couple `/operator` and `/pkg/complianceoperator`. As far as I can tell they are unrelated (apart from the name and similar tooling).
This target should probably go in the top-level `Makefile` instead.

**Review by porridge — DISMISSED** [review]
You might also be interested in the changes Malte is doing in https://github.com/stackrox/stackrox/pull/3012

File: operator/Makefile
Comment (janisz) [reply operator/Makefile]: Fixed

File: go.mod
Comment (porridge) [inline go.mod:398]: I thought with more recent go, this section is for `indirect` deps only? Won't `go mod tidy` kick it out?

File: Makefile
Comment (porridge) [inline Makefile:129]: I think we should pin the version. Or does it take the version from go.mod?

File: Makefile
Comment (porridge) [inline Makefile:303]: I'd prefer to have a separate copy of the boilerplate for complianceoperator, but I don't feel strongly about this.

File: Makefile
Comment (janisz) [reply Makefile:129]: It takes version from go.mod. I've created https://github.com/stackrox/stackrox/pull/3064 to move generators to separated go module like linters so they code be independent from our other deps.

File: go.mod
Comment (janisz) [reply go.mod]: Right, I moved it up

File: Makefile
Comment (janisz) [reply Makefile]: Done, I created empty file like was before

**Comment by janisz** [comment]
@porridge PTAL

File: go.mod
Comment (porridge) [inline go.mod:134]: We seem to be using `v0.8.0` for the main operator.
Just curious why you chose `v0.7.0` here? Just to make it clear: it's absolutely fine if there was an actual reason. Otherwise it might make sense to be consistent.

File: Makefile
Comment (porridge) [inline Makefile:305]: Since the target file can only provide functions for the `.../api/v1alpha1` package, I think it's safer to consider only those files in the actual commands?

```suggestion
COMPLIANCEOPERATOR_FILES = $(shell grep -R '+k8s:deepcopy-gen' pkg/complianceoperator/api/v1alpha1 -l)
pkg/complianceoperator/api/v1alpha1/zz_generated.deepcopy.go: $(CONTROLLER_GEN_BIN) $(COMPLIANCEOPERATOR_FILES) ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
	$(CONTROLLER_GEN_BIN) object:headerFile="tools/boilerplate.go.txt" paths="./pkg/complianceoperator/api/v1alpha1/..."
	# The generated source files might not comply with the current go formatting, so format them explicitly.
	go fmt ./pkg/complianceoperator/api/v1alpha1/...
```

**Review by porridge — APPROVED** [review]
Awesome work!

File: go.mod
Comment (janisz) [reply go.mod:134]: +1 I thought it uses newer k8s deps but I was checking v0.9 not v0.8. I'll fix it

**Comment by janisz** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/2992?utm_source=stack-comment).


Job Reference: y2dQx2xQ6xVtYgViFu9X

**Pull request merged**
